code/modules/research/xenobiology/crossbreeding/regenerative.dm

```diff
@@ -0,0 +1,245 @@
+
+/obj/item/slimecross/regenerative/metal
+	colour = "metal"
+
+/obj/item/slimecross/regenerative/metal/core_effect(mob/living/target, mob/user)
+	target.visible_message("<span class='warning'>The milky goo hardens and reshapes itself, encasing [target]!</span>")
+	var/obj/structure/closet/C = new /obj/structure/closet(target.loc)
+	C.name = "slimy closet"
+	C.desc = "Looking closer, it seems to be made of a sort of solid, opaque, metal-like goo."
+	target.forceMove(C)
+
+/obj/item/slimecross/regenerative/yellow
+	colour = "yellow"
+
+/obj/item/slimecross/regenerative/yellow/core_effect(mob/living/target, mob/user)
+	var/list/batteries = list()
+	for(var/obj/item/stock_parts/cell/C in target.GetAllContents())
+		if(C.charge < C.maxcharge)
+			batteries += C
+	if(batteries.len)
+		var/obj/item/stock_parts/cell/ToCharge = pick(batteries)
+		ToCharge.charge = ToCharge.maxcharge
+		to_chat(target, "<span class='notice'>You feel a strange electrical pulse, and one of your electrical items was recharged.</span>")
+
+/obj/item/slimecross/regenerative/darkpurple
+	colour = "dark purple"
+
+/obj/item/slimecross/regenerative/darkpurple/core_effect(mob/living/target, mob/user)
+	var/equipped = 0
+	equipped += target.equip_to_slot_or_del(new /obj/item/clothing/shoes/sneakers/purple(null), slot_shoes)
+	equipped += target.equip_to_slot_or_del(new /obj/item/clothing/under/color/lightpurple(null), slot_w_uniform)
+	equipped += target.equip_to_slot_or_del(new /obj/item/clothing/gloves/color/purple(null), slot_gloves)
+	equipped += target.equip_to_slot_or_del(new /obj/item/clothing/head/soft/purple(null), slot_head)
+	if(equipped > 0)
+		target.visible_message("<span class='notice'>The milky goo congeals into clothing!</span>")
+
+/obj/item/slimecross/regenerative/darkblue
+	colour = "dark blue"
+
+/obj/item/slimecross/regenerative/darkblue/core_effect(mob/living/target, mob/user)
+	if(!ishuman(target))
+		return
+	var/mob/living/carbon/human/H = target
+	var/fireproofed = FALSE
+	if(H.get_item_by_slot(slot_wear_suit))
+		fireproofed = TRUE
+		var/obj/item/clothing/C = H.get_item_by_slot(slot_wear_suit)
+		src.fireproof(C)
+	if(H.get_item_by_slot(slot_head))
+		fireproofed = TRUE
+		var/obj/item/clothing/C = H.get_item_by_slot(slot_head)
+		src.fireproof(C)
+	if(fireproofed)
+		target.visible_message("<span class='notice'>Some of [target]'s clothing gets coated in the goo, and turns blue!</span>")
+
+/obj/item/slimecross/regenerative/darkblue/proc/fireproof(obj/item/clothing/C)
+	C.name = "fireproofed [C.name]"
+	C.remove_atom_colour(WASHABLE_COLOUR_PRIORITY)
+	C.add_atom_colour("#000080", FIXED_COLOUR_PRIORITY)
+	C.max_heat_protection_temperature = FIRE_IMMUNITY_SUIT_MAX_TEMP_PROTECT
+	C.heat_protection = C.body_parts_covered
+	C.resistance_flags |= FIRE_PROOF
+
+/obj/item/slimecross/regenerative/silver
+	colour = "silver"
+
+/obj/item/slimecross/regenerative/cerulean/core_effect(mob/living/target, mob/user)
```

Comment:
uhh

Reply:
Pay no attention to the man behind the curtain!